src/ert/config/gen_data_config.py

```diff
@@ -1,21 +1,186 @@
+    ResponseDataInitialLayout,
+    ResponseTypes,
+)
 from ert.validation import rangestring_to_list
 
-from ._option_dict import option_dict
-from .parsing import ConfigValidationError, ErrorInfo
-from .response_config import ResponseConfig
-
 
 @dataclass
 class GenDataConfig(ResponseConfig):
+    @property
+    def primary_keys(self) -> List[str]:
+        return ["index", "report_step"]
+
+    @property
+    def response_type(self) -> str:
+        return ResponseTypes.GEN_DATA
+
+    @property
+    def data_layout(self) -> ResponseDataInitialLayout:
+        return ResponseDataInitialLayout.ONE_FILE_PER_NAME
+
+    @staticmethod
+    def _create_gen_obs(
+        scalar_value: Optional[Tuple[float, float]] = None,
+        obs_file: Optional[str] = None,
+        data_index: Optional[str] = None,
+    ) -> GenObservation:
+        if scalar_value is None and obs_file is None:
+            raise ValueError(
+                "Exactly one the scalar_value and obs_file arguments must be present"
+            )
+
+        if scalar_value is not None and obs_file is not None:
+            raise ValueError(
+                "Exactly one the scalar_value and obs_file arguments must be present"
+            )
+
+        if obs_file is not None:
+            try:
+                file_values = np.loadtxt(obs_file, delimiter=None).ravel()
+            except ValueError as err:
+                raise ObservationConfigError.with_context(
+                    f"Failed to read OBS_FILE {obs_file}: {err}", obs_file
+                ) from err
+            if len(file_values) % 2 != 0:
+                raise ObservationConfigError.with_context(
+                    "Expected even number of values in GENERAL_OBSERVATION", obs_file
+                )
+            values = file_values[::2]
+            stds = file_values[1::2]
+
+        else:
+            assert scalar_value is not None
+            obs_value, obs_std = scalar_value
+            values = np.array([obs_value])
+            stds = np.array([obs_std])
+
+        if data_index is not None:
+            indices = np.array([])
+            if os.path.isfile(data_index):
+                indices = np.loadtxt(data_index, delimiter=None, dtype=int).ravel()
+            else:
+                indices = np.array(
+                    sorted(rangestring_to_list(data_index)), dtype=np.int32
+                )
+        else:
+            indices = np.arange(len(values))
+        std_scaling = np.full(len(values), 1.0)
+        if len({len(stds), len(values), len(indices)}) != 1:
+            raise ObservationConfigError.with_context(
+                f"Values ({values}), error ({stds}) and "
+                f"index list ({indices}) must be of equal length",
+                obs_file if obs_file is not None else "",
+            )
+        return GenObservation(values, stds, indices, std_scaling)
+
+    @staticmethod
+    def parse_observation(args: ObsArgs) -> Dict[str, ObsVector]:
```

Comment:
All of the changes here are mostly just moving the code if I am not mistaken?

Reply:
Yep, that and adding the response type properties